packages/utils/src/schema/retrieveSchema.ts

```diff
@@ -516,8 +545,38 @@ export function retrieveSchemaInternal<
         return resolvedSchemaWithoutAllOf as S;
       }
     }
+    if (PROPERTIES_KEY in resolvedSchema && PATTERN_PROPERTIES_KEY in resolvedSchema) {
+      resolvedSchema = Object.keys(resolvedSchema.properties!).reduce(
+        (schema, key) => {
+          const patternProperties = Object.keys(schema.patternProperties!)
+            .filter((pattern) => RegExp(pattern).test(key))
+            .reduce(
+              (obj, pattern) => {
+                obj[pattern] = schema.patternProperties![pattern];
+                return obj;
+              },
+              {} as NonNullable<S['patternProperties']>,
+            );
```

Comment:
This looks like a duplicate of the code at lines 400-408, consider drying things up with a helper method